tests/ui/js/transformsTests.js

```diff
@@ -279,4 +279,139 @@ https://raw.githubusercontent.com/fluid-project/sjrk-story-telling/master/LICENS
         });
     });
 
+    var negateTransformTestCases = {
+        "test_01": {
+            input: true,
+            expectedResult: false,
+            expectedInverseResult: true
+        },
+        "test_02": {
+            input: false,
+            expectedResult: true,
+            expectedInverseResult: false
+        },
+        "test_03": {
+            input: 0,
+            expectedResult: true,
+            expectedInverseResult: false
+        },
+        "test_04": {
+            input: 1,
+            expectedResult: false,
+            expectedInverseResult: true
+        },
+        "test_05": {
+            input: -1,
+            expectedResult: false,
+            expectedInverseResult: true
+        },
+        "test_06": {
+            input: "",
+            expectedResult: true,
+            expectedInverseResult: false
+        },
+        "test_07": {
+            input: "Rootbeer",
+            expectedResult: false,
+            expectedInverseResult: true
+        },
+        "test_08": {
+            input: {},
+            expectedResult: false,
+            expectedInverseResult: true
+        },
+        "test_09": {
+            input: [],
+            expectedResult: false,
+            expectedInverseResult: true
+        },
+        "test_10": {
+            input: [1],
+            expectedResult: false,
+            expectedInverseResult: true
+        },
+        "test_11": {
+            input: [0],
+            expectedResult: false,
+            expectedInverseResult: true
+        },
+        "test_12": {
+            input: [""],
+            expectedResult: false,
+            expectedInverseResult: true
+        },
+        "test_13": {
+            input: "true",
+            expectedResult: false,
+            expectedInverseResult: true
+        },
+        "test_14": {
+            input: "false",
+            expectedResult: false,
+            expectedInverseResult: true
+        },
+        "test_15": {
+            input: "1",
+            expectedResult: false,
+            expectedInverseResult: true
+        },
+        "test_16": {
+            input: "0",
+            expectedResult: false,
+            expectedInverseResult: true
+        },
+        "test_17": {
+            input: "-1",
+            expectedResult: false,
+            expectedInverseResult: true
+        },
+        "test_18": {
+            input: null,
+            expectedResult: true,
+            expectedInverseResult: false
+        },
+        "test_19": {
+            // this case does not resolve as expected. normally, !undefined === true
```

Comment:
If the result of this test case is not expected, shall the transform be fixed to handle it?